src/stack_mitosis/predict.clj

```diff
@@ -77,6 +77,28 @@
             (update (lookup/position instances parent) detach child)
             (update (lookup/position instances child) promote))))))
 
+(defmethod predict :RestoreDBInstanceFromDBSnapshot
+  [instances op]
+  {:post [(lookup/exists? % (r/db-id op))]}
+  (let [source-id (->> op :meta :SourceDBInstance :DBInstanceIdentifier)
+        source-db (lookup/by-id instances source-id)
+        target-id (r/db-id op)
+        ;; We use the arn for some policy calculations, and we don't want it to
+        ;; end up being the source-arn
+        target-arn (-> source-db :DBInstanceArn (str/split #":db:") first (str ":db:" target-id))]
+    (-> op
+        :request
+        (dissoc :DBSnapshotIdentifier)
+        ;; This is not reeeally what happens, but replicating what happens is
+        ;; complicated. We're getting a DB in the new subnet, missing a few
+        ;; params, that are not going to be === the source params, but carry
+        ;; default values from RDS.
+        ;;
+        ;; Also, invert the param order so our params override source-db's
+        (#(merge %2 %1) source-db)
+        (assoc :DBInstanceArn target-arn)
+        (#(conj %2 %1) instances))))
```

Comment:
A little confused here, maybe just clojure styling or verb choice? I tend to avoid using the `#()` functions in a thread, but it looks like this might be a case where the `as->` thread would allow swapping the position of the hidden argument to make it clearer? Or it might just be that the conj at the end shouldn't be in the thread and it would be cleaner to create the instance attributes, and then as a second thread-last, merge those attributes into the source and conj it onto instances?

Anyway I will take a closer look later.

Reply:
`as->` is cool! Gave it a try

Reply:
Awesome, thank you, that's much nicer. The threading macros are really nice, but when the direction flips back and forth they are a bit more difficult to follow, so the `as->` really helps here.